enclave/src/rpc/rpc_get_balance.rs

```diff
@@ -0,0 +1,162 @@
+// This file is part of Polkadex.
+
+// Copyright (C) 2020-2021 Polkadex oü.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+pub extern crate alloc;
+use alloc::{boxed::Box, string::String, string::ToString};
+
+use crate::polkadex_balance_storage::Balances;
+use crate::rpc::polkadex_rpc_gateway::RpcGateway;
+use crate::rpc::rpc_call_encoder::{JsonRpcCallEncoder, RpcCall, RpcCallEncoder};
+use crate::rpc::rpc_info::RpcCallStatus;
+use crate::rpc::trusted_operation_verifier::TrustedOperationExtractor;
+use jsonrpc_core::{BoxFuture, Params, Result as RpcResult, RpcMethodSync, Value};
+use log::*;
+use polkadex_sgx_primitives::types::DirectRequest;
+use substratee_stf::{Getter, TrustedGetter, TrustedOperation};
+use substratee_worker_primitives::DirectRequestStatus;
+
+pub struct RpcGetBalance {
+    top_extractor: Box<dyn TrustedOperationExtractor + 'static>,
+    rpc_gateway: Box<dyn RpcGateway + 'static>,
+}
```

Comment:
Use trait objects inside RpcGetBalance to promote de-coupling and allow clean unit testing